desc/plotting.py

```diff
@@ -3753,3 +3760,341 @@ def plot_logo(save_path=None, **kwargs):
+        * ``xlabel_fontsize``: float, fontsize of the xlabels
+        * ``ylabel_fontsize``: float, fontsize of the ylabels
+
+
+    Outputs
+    -------
+    figdata : dict of matplotlib.figure.Figure
+        Dictionary with the keys ``"fig_X"`` with values ``matplotlib.figure.Figure``,
+        with ``X`` corresponding to the different figure types described in the
+        docstring description, which are the figures being plotted to.
+    axdata : dict
+        Dictionary with the keys ``"ax_X"`` with values ``matplotlib.axes.Axes``
+        or ``ndarray`` of ``matplotlib.axes.Axes``, with ``X`` corresponding to the
+        different figure types described in the docstring description, which are
+        the axes being plotted to.
+    plot_data : dict
+        dictionary of the data plotted, only returned if ``return_data=True``
+        This is the same as data_regcoil
+
+    """
+    try:
+        # if it is a list, just grab the first one
+        # as we change the attribute anyways so just need the correct
+        # geometry
+        field = field[0]
+    except TypeError:
+        # it was not a list, so proceed as usual
+        pass
+    cmap = kwargs.pop("cmap", "viridis")
+    field = (
+        field.copy()
+    )  # copy the field so that we are not changing the passed-in field
+    # TODO: check that field has correct NFP and resolutions?
+    scan = len(data["Phi_mn"]) > 1
+    scan_str = "_scan" if scan else ""
+
+    lambdas = data["lambda_regularization"]
+    chi2Bs = data["chi^2_B"]
+    chi2Ks = data["chi^2_K"]
+    phi_mns = data["Phi_mn"]
+    if eval_grid is None:
+        eval_grid = data["eval_grid"]
+        eval_grid = (
+            eval_grid
+            if (not eval_grid.sym and eval_grid.N != 0)
+            else LinearGrid(
+                M=eval_grid.M,
+                N=max(eval_grid.N, 1),
+                sym=False,
+                NFP=eval_grid.NFP,
+                endpoint=True,
+            )
+        )
+    if source_grid is None:
+        source_grid = data["source_grid"]
+
+    # check if we can use existing quantities in data
+    # or re-evaluate based off of the new grids passed-in
+    recalc_eval_grid_quantites = not eval_grid.equiv(
+        data["eval_grid"]
+    ) or not source_grid.equiv(data["source_grid"])
+
+    if "external_field" in data.keys() and recalc_eval_grid_quantites:
+        external_field = data["external_field"]
+    else:
+        external_field = None
+
+    ncontours = kwargs.pop("ncontours", 15)
+    markersize = kwargs.pop("markersize", 12)
+    xlabel_fontsize = kwargs.pop("xlabel_fontsize", 24)
+    ylabel_fontsize = kwargs.pop("ylabel_fontsize", 24)
+    title_fontsize = kwargs.pop("title_fontsize", 24)
+
+    figdata = {}
+    axdata = {}
+    # show composite scan over lambda_regularization plots
+    # strongly based off of Landreman's REGCOIL plotting routine:
+    # github.com/landreman/regcoil/blob/master/
+    figsize = kwargs.pop("figsize", (16, 12))
```

Comment:
another reason to remove or break this up: fig/font size that makes sense for a single scatter plot don't work for the big multi-contour plots, and vice versa. The default settings make the scatter plots huge with tiny tick labels (see notebook)

Reply:
this function might be better as just a few small helper functions defined in the notebook